assets/js/components/KeyMetrics/ConnectGA4CTAWidget.js

```diff
@@ -0,0 +1,208 @@
+			'analytics',
+			true
+		)
+	);
+	const settingsURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
+	);
+	const isAnalyticsConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics' )
+	);
+	const connectGA4URL = `${ settingsURL }#connected-services/analytics/edit`;
+	const isNavigatingToGA4URL = useSelect( ( select ) =>
+		select( CORE_LOCATION ).isNavigatingTo( connectGA4URL )
+	);
+
+	const { dismissItem } = useDispatch( CORE_USER );
+	const { setValues } = useDispatch( CORE_FORMS );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	const activateAnalytics = useActivateModuleCallback( 'analytics' );
+	const completeAnalyticsActivation =
+		useCompleteModuleActivationCallback( 'analytics' );
+
+	const handleCTAClick = useCallback( async () => {
+		if ( ! isAnalyticsConnected ) {
+			if ( isAnalyticsActive ) {
+				completeAnalyticsActivation();
+			} else {
+				activateAnalytics();
+			}
+		} else if ( ! isGA4Connected ) {
+			setValues( FORM_SETUP, {
+				// Pre-enable GA4 controls.
+				enableGA4: true,
+				// Enable tooltip highlighting GA4 property select.
+				enableGA4PropertyTooltip: true,
+			} );
+
+			await snapshotAllStores();
+
+			navigateTo( `${ settingsURL }#connected-services/analytics/edit` );
+		}
+	}, [
+		activateAnalytics,
+		completeAnalyticsActivation,
+		isAnalyticsActive,
+		isAnalyticsConnected,
+		isGA4Connected,
+		navigateTo,
+		setValues,
+		settingsURL,
+	] );
+
+	const [ inProgress, setInProgress ] = useState( false );
+
+	/*
+	 * Using debounce here because the spinner has to render across two separate calls.
+	 * Rather than risk it flickering on and off in between the activation call completing and
+	 * the navigate call starting, we will just set a debounce to keep the spinner for 3 seconds.
+	 */
+	const debouncedSetInProgress = useDebounce( setInProgress, 3000 );
+
+	useEffect( () => {
+		if (
+			isActivatingAnalytics ||
+			isNavigatingToReauthURL ||
+			isNavigatingToGA4URL
+		) {
+			setInProgress( true );
+		} else {
+			debouncedSetInProgress( false );
+		}
+	}, [
+		isActivatingAnalytics,
+		isNavigatingToReauthURL,
+		debouncedSetInProgress,
+		isNavigatingToGA4URL,
+	] );
+
+	if (
+		isCTADismissed ||
+		! isUserInputCompleted ||
+		isGA4Connected ||
+		ga4DependantKeyMetrics.length < 3
+	) {
+		return <WidgetNull />;
+	}
+
+	return (
+		<Widget
+			noPadding
+			Footer={ () => (
+				<KeyMetricsCTAFooter
+					onActionClick={ () => dismissItem( DISMISSED_ITEM_KEY ) }
+				/>
+			) }
+		>
+			<KeyMetricsCTAContent
+				className="googlesitekit-km-connect-ga4-cta"
+				title={ __(
+					'Google Analytics is disconnected',
+					'google-site-kit'
+				) }
+				description={ __(
+					'Metrics cannot be displayed without Google Analytics',
+					'google-site-kit'
+				) }
+				actions={
+					<SpinnerButton
+						onClick={ handleCTAClick }
+						isSaving={ inProgress }
```

Comment:
Please also make the button `disabled` when saving. This should happen by default IMO but it doesn't work that way currently.

Reply:
Excellent spot, updated, thanks!